tests/unit/array/storage/weaviate/test_query_params.py

```diff
@@ -2,31 +2,32 @@
 import numpy as np
 
 
+def find_random(da, target_certainty):
+    return da.find(
+        DocumentArray([Document(embedding=np.random.randint(10, size=10))]),
+        query_params={"certainty": target_certainty},
+    )[0]
+
+
 def test_certainty_filter(start_storage):
     nrof_docs = 100
+    target_certainty = 0.98
     da = DocumentArray(
         storage="weaviate", config={"name": "Test", "host": "localhost", "port": 8080}
     )
 
     with da:
         da.extend(
             [
-                Document(id=f'r{i}', embedding=np.ones((3,)) * i)
-                for i in range(nrof_docs)
+                Document(id=f"r{i}", embedding=np.random.randint(10, size=10))
+                for i in range(1, nrof_docs)
             ],
         )
 
-    results = da.find(
-        DocumentArray([Document(embedding=[1, 2, 3])]),
-        query_params={"certainty": 0.9}
-    )
+    while True:
```

Comment:
why not initialize `results = []` and do `while len(results) == 0`? I panic a little about `while True`

Reply:
I tested that a bunch of times because I was a bit nervous but i can alter it. i just need some return to do the check.

Reply:
just updated the code and tested it
its pushing now